RecoEcal/EgammaCoreTools/interface/MustacheSCParametersHelper.h

```diff
@@ -0,0 +1,25 @@
+#ifndef RecoEcal_EgammaCoreTools_MustacheSCParametersHelper_h
+#define RecoEcal_EgammaCoreTools_MustacheSCParametersHelper_h
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/EcalObjects/interface/EcalMustacheSCParameters.h"
+
+namespace reco {
+
+  class MustacheSCParametersHelper : public EcalMustacheSCParameters {
```

Comment:
I find this pattern somewhat confusing. To be more precise
* the `static_cast<EcalMustacheSCParameters>()` in the ESProducer looks weird (even if it is technically correct)
* the creation of `MustacheSCParametersHelper` from `EcalMustacheSCParameters` leads to copying all the data

Would it be feasible to move `sqrtLogClustETuning()` and `parabolaParameters()` to `EcalMustacheSCParameters` directly?

If not, how about making `MustacheSCParametersHelper` to hold `EcalMustacheSCParameters` by (`const`) pointer to avoid the copying? (the helper could even be produced by and ESProducer to avoid repeating its construction in different modules and EDM streams).

Reply:
The intention was that the `EcalMustacheSCParameters` in CondFormats` holds the data and all manipulations of the data are done with the `MustacheSCParametersHelper` in the `RecoEcal` package.

Reply:
@thomreis I think the principle you describe makes sense to me, however, I'm trying to understand if the implementation could perhaps be simplified further.

Is the only functionality of the helper to prepare the data `EcalMustacheSCParameters` from the config using `MustacheSCParametersHelper::MustacheSCParametersHelper(const edm::ParameterSet &iConfig)`?

If so, can you remove the helper class entirely and just use an utility function located in e.g. RecoEcal/EgammaCoreTools to prepare EcalMustacheSCParameters in the ESProducer?

Reply:
The `EcalMustacheSCParameters` has only getter functions but no setters at the moment. Initially, even the getters were in the helper but this was changed to avoid copying the objects.
This makes changes on the ECAL side possible (e.g. changes number of parameters in case we want to use a higher order polynomial at some point) without touching the code in the CondFormats directory.
Having a utility function to prepare the conditions from the config file would mean to include setters in `EcalMustacheSCParameters` which would make it possible to set the conditions directly without the checks and sorting that the helper does. Alternatively, the helper could be made a friend class in RecoEcal could contain the setters.

Reply:
To me a `friend` class would be clearer pattern than using inheritance for a deriving class to set members of the base class and then slice the base part of the object with `static_cast`.

A `friend` would be more restrictive, now anyone could fill the contents of the base class in arbitrary way by inheriting with a custom class (even if that could be unlikely for anyone to do in practice).

Reply:
I have implemented the `friend` solution now.